### PR TITLE
Added top_k_accuracy

### PR DESCRIPTION
**Pull request by tonydavis629**
## Description

Added scikit learn top_k_accuracy metric to score_function.py metrics

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by tonydavis629** [comment]
I removed the wrapper function and checked yapf and flake8, but I'm getting an error on flake8 that top_k_accuracy_score is imported but not used. I assume this is ok?

File: deepchem/metrics/score_function.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from sklearn.metrics import roc_auc_score  # noqa
 from sklearn.metrics import accuracy_score  # noqa
 from sklearn.metrics import balanced_accuracy_score  # noqa
+from sklearn.metrics import top_k_accuracy_score
```
Comment (rbharath) [inline deepchem/metrics/score_function.py:19]: You want to add # noqa here:

```
from sklearn.metrics import top_k_accuracy_score  # noqa
```
Hopefully that fixes the flake8 issue

**Comment by rbharath** [comment]
Looks like there's still a yapf error. Make sure you have the right version of yapf locally (0.22)

**Comment by rbharath** [comment]
@tonydavis629 It looks like yapf/flake8 are still failing. See https://github.com/deepchem/deepchem/runs/4589451595?check_suite_focus=true

I'd recommend double checking that yapf is version 0.22 locally. Happy holidays!

**Comment by tonydavis629** [comment]
Ah I was mistakenly running yapf on my main branch, this should do it.

**Comment by tonydavis629** [comment]
Hi Bharath the yapf and flake8 issues look to be passing, can we merge this?

**Comment by rbharath** [comment]
@tonydavis629 Can you rebase the PR? I think you're right, but there are a lot of old CI failures that are making it hard to verify (we should have ~32 passing CI runs right now). If you do the rebase, we should get a fresh run on the latest code

**Comment by tonydavis629** [comment]
Ok I have rebased it, looks like it worked out.

**Comment by rbharath** [comment]
@tonydavis629 I believe you did a pull and not a rebase. There are a bunch of commits and files modified here that don't have to do with our PR (notice the 70 files changed). It might be easier to just open up a fresh PR and copy over the changes manually. I'm also glad to talk through the rebase process on our next conversation offline

**Comment by tonydavis629** [comment]
Ah ok I wasn't sure if the changed files were expected or not. I will start a new PR.
